ide/commons-gwt/src/main/java/org/eclipse/che/ide/util/PathEncoder.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2012-2017 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.ide.util;
+
+import static com.google.gwt.http.client.URL.encodePathSegment;
+import static org.eclipse.che.ide.resource.Path.SEPARATOR;
+
+import org.eclipse.che.ide.resource.Path;
+
+/**
+ * @author Alexander Andrienko
+ * @author Mykola Morhun
+ */
+public class PathEncoder {
```

Comment:
Add several tests also.

Reply:
Well, we have only one method to test and it uses native javascript method. I think it won't worst to try to deal with it in tests.

Reply:
But I tried...

Reply:
It is pure java code. Why do you need js?

Reply:
Look deeper. `encodeComponent` method uses native js method under the hood.